app/models/favorites.rb

```diff
@@ -0,0 +1,15 @@
+class Favorites
+  attr_reader :contents
+
+  def initialize(initial_contents)
+    @contents = initial_contents || Hash.new
```

Comment:
If you try to initialize with a non-existing session[:favorites], it will create a new hash